File: php/src/Model/WorkerTimeEntriesResult.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * The recorded shift and break log for one worker over a window — the working-time record an EU
 * employer must keep (CJEU C-55/18).
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkerTimeEntriesResult
{
    public function __construct(
        public readonly string $workerId,
        public readonly string $from,
        public readonly string $to,
        /** @var list<WorkerTimeEntry> */
        public readonly array $entries,
        public readonly WorkerTimeTotals $totals,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            workerId: (string) ($data['workerId'] ?? ''),
            from: (string) ($data['from'] ?? ''),
            to: (string) ($data['to'] ?? ''),
            entries: Json::parseEach($data, 'entries', [WorkerTimeEntry::class, 'fromArray']),
            totals: WorkerTimeTotals::fromArray($data['totals'] ?? []),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/WorkerTimeEntriesResult.java
package io.fivexer.sdk.model;

import java.util.List;

/**
 * The recorded shift and break log for one worker over a window — the working-time record an
 * EU employer must keep (CJEU C-55/18).
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkerTimeEntriesResult {
    private String workerId;
    private String from;
    private String to;
    private List<WorkerTimeEntry> entries;
    private WorkerTimeTotals totals;

    public String getWorkerId() { return workerId; }
    public String getFrom() { return from; }
    public String getTo() { return to; }
    public List<WorkerTimeEntry> getEntries() { return entries; }
    public WorkerTimeTotals getTotals() { return totals; }
}
